backend/contracts/ElectionFactory.sol

```diff
@@ -1,28 +1,51 @@
-//SPDX-License-Identifier: UNLICENSED
+// SPDX-License-Identifier: UNLICENSED
 
 pragma solidity ^0.8.0;
 
 contract ElectionFactory {
-    struct Election {
+    struct Candidate {
+        uint8 id;
         string name;
     }
 
-    Election[] public elections;
-
-    constructor() {
+    struct Ballot {
+        string name;
+        string description;
+        uint256 createTime;
+        uint256 votingTime;
+        Candidate[] candidates;
     }
 
-    function createElection(string memory _name) external{
-        Election memory newElection = Election(_name);
-        elections.push(newElection);
+    mapping(uint256 => Ballot) public ballots;
+    uint256 public nextBallotId;
+
+    event CandidateAdded(uint256 ballotId, uint8 id, string name);
+    event ContractInitialized(uint256 ballotId, string name, string description, uint256 createTime, uint256 votingTime, Candidate[] candidates);
```

Comment:
What is the use case for events here?

Reply:
I considered these events could be useful for ballot success creation control on js backend. Should I remove them?

Reply:
On the backend, we can be certain that a certain transaction has run successfully if no error has been received in the execution process. So these events are obsolete if that was the idea behind them.

Reply:
alright then, will remove them too